### ISSUE TITLE
What does SmtpError IpBlacklisted mean exactly?

### ISSUE THREAD

**Issue by Boscop** [reacherhq/check-if-email-exists#1378]
Thanks for making this useful tool :)
When I run it from my local computer, for some emails I get SmtpError IpBlacklisted, e.g.:
```json
  "smtp": {
    "error": {
      "type": "SmtpError",
      "message": "permanent: zen.mimecast.org https://www.spamhaus.org/query/ip/77.85.116.89. - https://community.mimecast.com/docs/DOC-1369#550 [UioxhvvdMDyD8YY9lHrBjQ.us77]"
    },
    "description": "IpBlacklisted"
  },
```
But that IP (77.85.116.89) is not my IP. 
So I'm wondering what this means? (The `is_reachable` field is `"unknown"` in that case, so it seems I'm blacklisted even though this is not my IP.)
Why does this happen?

**Comment by amaury1729** [comment]
The IP is the incoming request's IP that the destination server sees. Are you using a proxy? Are you being a VPN?

**Comment by Boscop** [comment]
@amaury1729 No, I'm not using a proxy or VPN, I'm running it from my laptop at home. Any idea why my residential IP is blacklisted? :)

**Comment by amaury1729** [comment]
In general, ISPs block outgoing port 25, so I don't recommend verifying emails using residential IPs. That IP might belong to your ISP?

In any case, I would recommend running it with a fixed server IP, so that you have some control over its reputation. In your case, if you'd like to deep digger, you can paste the logs after running RUST_LOG=debug, I can maybe try to help.
